### ISSUE TITLE
Number() with pipes formatting is bad

### ISSUE THREAD

**Issue by rentalhost** [prettier/prettier#14754]
**Prettier 2.8.8**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6VACAQgQQCIYDCA8ngKJIA6UcAHgA4QBOMGkUAzq9dWAIb0+AIwCWAGxEwRcDlj4c4GALwZqGdRoByAVwC2QuEwAUaogOHjJ0jgDoSmzWUIAVAPoB1AJLOAEq7xYGAA+phoa1ITmohJSMnYAMp5kmm4ACgBKxM7EJPGuACwAjMGhYapQZRqRgtFWcQDKTgCq6WSu9o4unsSapZX9YQCUANzUIAA0IBD0UtAcyKB8TEwQAO6pSwjzKHxiq3wAnvOTQkx8YADWcDD1fLpw8SI0yABmuwonZ5fX9YJgTwBzZAwJjaOCTOD6OAAE2hMPifCgAO0fABcAAYsxdHwYFIkcgQHxtDAIBMQAALGC6MTucmSGR-OD1LZWABukgOBLAHGOICeChYqTOAOxr3e4JAACsOLR6oCxHAAIraCDwMViD4gQRMAUEoTCOBiMn0JhPGDuETQmDk5AADgADJMTRAFO4zvQCSaZIZWXAyQBHFXwIXTbaEjgAWhoMJhZKYcEDInjQtRoqQbw1EoUuhEwNBWflcBwuNNQmJcFShkez3T4smHELytVftrmcmMGEFqtNqQACZ22dLEjCBBdGmQDIAKxk7QKZzCbYZzWssGeKBw2D1MCmmY4df1GAHBXqhQAX1PQA)

**Input:**

```jsx
// BAD CODE:
export const 

capabilitiesBase = 
      Number(
  Capabilities.CONNECT_WITH_DB |
        
Capabilities.CLIENT_PROTOCOL_41 |
        
        Capabilities.SECURE_CONNECTION
                     );
```

**Output:**

```jsx
// BAD CODE:
export const capabilitiesBase = Number(
  Capabilities.CONNECT_WITH_DB |
    Capabilities.CLIENT_PROTOCOL_41 |
    Capabilities.SECURE_CONNECTION
);
```

**Expected output:**

```jsx
// BAD CODE:
export const capabilitiesBase = Number(
  Capabilities.CONNECT_WITH_DB |
  Capabilities.CLIENT_PROTOCOL_41 |
  Capabilities.SECURE_CONNECTION
);
```

**Comment by sosukesuzuki** [comment]
It's not a bug. That indentation is added to make it clearly visible how many arguments the call has. Related: https://github.com/prettier/prettier/issues/4338, https://github.com/prettier/prettier/issues/6442

**Comment by alexander-akait** [comment]
@sosukesuzuki I think it makes sense to not indent for binary operators, becaue it is often bitmaps and better to read without indent

**Comment by sosukesuzuki** [comment]
You are right when expression is simple bitmaps, but I think it makes sense when it is in logical expr in ternary:

```js
fn(
  bifornCringerMoshedPerplexSawder,
  askTrovenaBeenaDependsRowans,
  glimseGlyphsHazardNoopsTieTie === averredBathersBoxroomBuggyNurl  &&
    anodyneCondosMalateOverateRetinol 
      ? annularCooeedSplicesWalksWayWay 
      : kochabCooieGameOnOboleUnweave 
);
```

So, even if we decide to improve formatting binary expr, we have to consider some cases.

**Comment by alexander-akait** [comment]
@sosukesuzuki I mean only for `Bitwise Operations`, i.e. `&`, `|` and etc, not for `&&`,  `||` or `... ? ... : ...` and etc

**Comment by sosukesuzuki** [comment]
I personally don't use bitmaps so I don't have an opinion, but I think it could be improved.

**Comment by thorn0** [comment]
@alexander-akait What if the second argument is added to the call? Will Prettier have to switch back to the current logic?

```js
// proposed (?):
export const capabilitiesBase = Number(
  Capabilities.CONNECT_WITH_DB |
  Capabilities.CLIENT_PROTOCOL_41 |
  Capabilities.SECURE_CONNECTION,
  secondArg
);
// current:
export const capabilitiesBase = Number(
  Capabilities.CONNECT_WITH_DB |
    Capabilities.CLIENT_PROTOCOL_41 |
    Capabilities.SECURE_CONNECTION,
  secondArg
);
```

**Comment by alexander-akait** [comment]
@thorn0 I prefer:
```js
// proposed (?):
export const capabilitiesBase = Number(
  Capabilities.CONNECT_WITH_DB |
  Capabilities.CLIENT_PROTOCOL_41 |
  Capabilities.SECURE_CONNECTION,
  secondArg
);
```

but it's very subjective, I use this in rust and js

**Comment by thorn0** [comment]
@alexander-akait Yes, very subjective

On the other hand, reformatting when the 2nd argument is added is diff-unfriendly. A tradeoff could be to implement the proposed formatting only for single-argument calls and only if all the operands are member expressions with the same head like `Capabilities.` in the examples here. But I'm not sure it's really worth changing. BTW, @rentalhost why do you need to cast a number to number? Or is it just an artificial example?

**Comment by rentalhost** [comment]
@thorn0 In my case, `Capabilities` has `bigint`, but I need the `const` to actually be a `number`.

**Comment by fisker** [comment]
~Out of curiosity, why `Number()`, shouldn't it already be a number?~ Sorry, already answered in https://github.com/prettier/prettier/issues/14754#issuecomment-1537186303

We already removed indentation for logical expressions in `Boolean()`, I think we can apply the same logic to `Number()`~, but the `Number()` type coercion seems unnecessary to me~.
